lib/domain/usecase/room/update_group_chat_interactor.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/room/update_group_chat_failure.dart';
+import 'package:fluffychat/domain/app_state/room/update_group_chat_loading.dart';
+import 'package:fluffychat/domain/app_state/room/update_group_chat_success.dart';
+import 'package:matrix/matrix.dart';
+
+class UpdateGroupChatInteractor {
+  Stream<Either<Failure, Success>> execute({
+    required Room room,
+    MatrixFile? avatar,
+    String? displayName,
+    String? description,
+    bool isDeleteAvatar = false,
+  }) async* {
+    yield const Right(UpdateGroupChatLoading());
+    try {
+      Logs().d(
+        'UpdateGroupChatInteractor::execute(): Avatar - ${avatar?.name} - displayName - $displayName - description - $description',
+      );
+
+      if (isDeleteAvatar) {
+        await room.setAvatar(null);
+      } else if (avatar != null) {
+        await room.setAvatar(avatar);
+      }
+
+      if (displayName != null) {
+        await room.setName(
+          displayName,
+        );
+      }
+
+      if (description != null) {
+        await room.setDescription(
+          description,
+        );
+      }
```

Comment:
you can use `Future.wait` to do all works in parallel

Reply:
not fixed yet

Reply:
If works in parallel, how to handle error?

Reply:
add onError for each future, so you can override default error

Reply:
I need to identify a specific error, not a general error. This is the reason why I have to make funcs synchronous.